src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,127 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final double GROW_MULTIPLYER = 1.5;
+    private static final int DEFAUL_SIZE = 10;
+    private int size = 0;
+    private int currentMaxSize;
+    private T[] elements;
+
+    public ArrayList() {
+        currentMaxSize = DEFAUL_SIZE;
+        this.elements = (T[]) new Object[DEFAUL_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == currentMaxSize) {
+            increaseArrCapacity();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of array boundaries");
+        }
+        if (size == currentMaxSize) {
+            increaseArrCapacity();
+        }
+        if (index != size) {
+            T previousElem = elements[index];
+            T nextElem;
+            elements[index] = value;
+            for (int i = index + 1; i < size + 1; i++) {
+                nextElem = elements[i];
+                elements[i] = previousElem;
+                previousElem = nextElem;
+            }
+        } else {
+            elements[index] = value;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        while (size + list.size() > currentMaxSize) {
+            increaseArrCapacity();
+        }
+        for (int i = 0; i < list.size(); i++) {
```

Comment:
```suggestion
        for (int i = 0; i < list.size(); i++) {
        resize();
```

Reply:
resize isn't needed here, Add all is functionally equivalent to calling add multiple times in a row. One of these add calls will trigger resize, when it will be needed

Reply:
You are absolutely right)